autosubmit/autosubmit.py

```diff
@@ -3486,61 +3486,54 @@ def configure(advanced, database_path, database_filename, local_root_path, platf
                     Log.error("jobs.yml path does not exist.")
                     return False
 
-            if machine:
-                rc_path = '/etc'
-            elif local:
-                rc_path = '.'
-            else:
-                rc_path = home_path
-            rc_path = rc_path.joinpath('.autosubmitrc')
+            rc_path: Path = get_rc_path(machine, local)
 
-            config_file = open(rc_path, 'w')
-            Log.info("Writing configuration file...")
-            try:
-                parser = ConfigParser()
-                parser.add_section('database')
-                parser.set('database', 'path', str(database_path))
-                if database_filename is not None and len(str(database_filename)) > 0:
-                    parser.set('database', 'filename', str(database_filename))
-                parser.add_section('local')
-                parser.set('local', 'path', str(local_root_path))
-                if (jobs_conf_path is not None and len(str(jobs_conf_path)) > 0) or (
-                        platforms_conf_path is not None and len(str(platforms_conf_path)) > 0):
-                    parser.add_section('conf')
-                    if jobs_conf_path is not None:
-                        parser.set('conf', 'jobs', str(jobs_conf_path))
-                    if platforms_conf_path is not None:
-                        parser.set('conf', 'platforms', str(platforms_conf_path))
-                if smtp_hostname is not None or mail_from is not None:
-                    parser.add_section('mail')
-                    parser.set('mail', 'smtp_server', smtp_hostname)
-                    parser.set('mail', 'mail_from', mail_from)
-                parser.add_section("globallogs")
-                parser.set("globallogs", "path", str(global_logs_path))
-                parser.add_section("structures")
-                parser.set("structures", "path", str(structures_path))
-                parser.add_section("historicdb")
-                parser.set("historicdb", "path", str(historicdb_path))
-                parser.add_section("historiclog")
-                parser.set("historiclog", "path", str(historiclog_path))
-                parser.add_section("autosubmitapi")
-                parser.set("autosubmitapi", "url", autosubmitapi_url)
-                # parser.add_section("hosts")
-                # parser.set("hosts", "whitelist", " localhost # Add your machine names")
-                parser.write(config_file)
-                config_file.close()
```

Comment:
This file was closed only in the success path. Harmless to leave it open as the exceptions should crash the interpreter and close everything; but with NFS, GPFS, it doesn't hurt to make sure the file will be really closed. Thus, this change that's moving all the lines to the right...